File: 12.6/john.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;


bool john_wins(const vector<int>& eminems) {
    int xor_sum = 0;
    for (int count : eminems) {
        xor_sum ^= count;
    }
    return xor_sum != 0;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int T;
    cin >> T;

    while (T--) {
        int N;
        cin >> N;

        vector<int> eminems(N);
        for (int i = 0; i < N; ++i) {
            cin >> eminems[i];
        }

        cout << (john_wins(eminems) ? "John" : "Brother") << '\n';
    }

    return 0;
}

File: 12.6/john.py
import sys


def john_wins(eminems):
    total_eminems = sum(eminems)
    if total_eminems == 1:
        return False

    xor_sum = 0
    for count in eminems:
        xor_sum ^= count

    return xor_sum != 0


def read_lines(T):
    lines = []
    for _ in range(T * 2):
        lines.append(sys.stdin.readline().strip())
    return lines


if __name__ == "__main__":
    T = int(sys.stdin.readline().strip())
    for _ in range(T):
        _ = int(sys.stdin.readline().strip())
        eminems = list(map(int, sys.stdin.readline().strip().split()))
        print("John" if john_wins(eminems) else "Brother")